### DISCUSSION TITLE
How to handle blocking calls in a handler?

### DISCUSSION THREAD

**Discussion by overthetop** [ntex-rs/ntex#438] [Q&A]
Hey all
What is the best practice to handle blocking calls in a http handler?
Is this the way to do it https://github.com/ntex-rs/examples/blob/a3cb7646348f0ba25efffc639125eec766860368/async_db/src/db.rs#L49 using  web.block( .... )? I suppose that is using another thread pool. How can I configure the number of threads that handle blocking calls? Please correct me and explain in case I'm wrong.

**Comment by fafhrd91** [comment]
ntex uses runtime’s blocking system.

for example for tokio ntex uses [spawn_blocking](https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html) you need to check target runtime docs

**Reply by overthetop** [reply]
Yes, but how to configure the blocking threads number for tokio? Isn't that exposed through the ntex interface?

**Reply by fafhrd91** [reply]
you have to create custom runtime

https://docs.rs/ntex/2.5.0/ntex/rt/struct.Builder.html#method.block_on

something like
```rust

fn main() {
   ntex::rt::Builder::new("app").block_on(|fut| {
       let rt = tokio::runtime::Builder::new_current_thread()
          .max_blocking_threads(512)
          .build().unwrap();
       rt.block_on(fut)
   })
   .finish()
   .run(my_app())
}

async my_app() {
   // some ntex code
}
```

do you really need more threads? default is 512

**Reply by overthetop** [reply]
Yes. Majority of the API calls will be blocking and I need to make sure the app can handle the bursts. Thanks, I'll give it a try and write back.
